### ISSUE TITLE
[FF2] Stack Overflow when overridding @Atomic methods

### ISSUE THREAD

**Issue by jcarvalho** [fenix-framework/fenix-framework#58]
Consider the following example:

``` java
class X {
    @Atomic
    public void doStuff() {
        (...)
    }
}

class Y extends X {
    @Atomic
    @Override
    public void doStuff() {
        (...)
        super.doStuff();
    }
}
```

Fenix Framework will rewrite both methods, renaming them both to atomic$doStuff (in their respective classes).

However, when a program calls `Y.doStuff()`, it will call the original method, now `Y.atomic$doStuff`, inside a transactional context.
That method, calls `super`, which in turn, will invoke X's `atomic$doStuff`, which, is also present in Y, and due to Java's dynamic dispatch on the receiver, Y's `atomic$doStuff` will be invoked, causing an infinite loop.

In short, here is a sample of the call stack

`doStuff() -> atomic$doStuff() -> super.doStuff() -> atomic$doStuff() 
-> super.doStuff() -> (...)`
